File: CPE/Exam/230321/pC_UVA-13055.py
"""
    Stack
"""
import sys
input = sys.stdin.readline
def print(val):
    sys.stdout.write(str(val) + '\n')

n = int(input())
st = []
for _ in range(n):
    cmd = input().split()
    if cmd[0] == "Sleep":
        st.append(cmd[1])
    elif cmd[0] == "Kick":
        if st:
            st.pop()
    else:
        if st:
            print(st[-1])
        else:
            print("Not in a dream")

File: CPE/Exam/230321/pC_UVA-13055.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int t;
    cin >> t;
    string s;
    stack<string> st;
    while (t--) {
        cin >> s;
        if (s == "Sleep") { // Sleep {name}
            cin >> s;
            st.push(s);
        } else if (s == "Kick") { // Kick
            if (!st.empty()) {
                st.pop();
            }
        } else { // Test
            if (!st.empty()) {
                cout << st.top() << endl;
            } else {
                cout << "Not in a dream" << endl;
            }
        }
    }
    return 0;
}